### PR TITLE
Better default keymap prefix Ctrl-Alt → Ctrl-i

### PR DESCRIPTION
**Pull request by ulidtko**
The default keymap is both awkward and dull.

For one, three-key combos like Ctrl-Alt-k are difficult enough to type "in-flow". Shortcuts supposed to be short!

Next, Ctrl-Alt-T is used in wide variety of Desktop Environments to **open new terminal** (e.g. #159) regardlessly of current focused app. Ctrl-Alt-L is used to lock screen, again globally. I could gratuitously extrapolate this up to Ctrl-Alt-Delete — and infer the convention that "ctrl+alt namespace" is for system shortcuts, not for apps.

What I propose in this change is very simple, and I really like that Atom can do this. Remember the <kbd>Ctrl-K</kbd> shortcuts in Atom, like <kbd>Ctrl-K</kbd><kbd>U</kbd> for uppercasing, <kbd>Ctrl-K</kbd><kbd>B</kbd> for show/hiding the treeview? We can do the same with <kbd>Ctrl-I</kbd>, and carve out a sort of "Idris namespace" in the keymap.

What this means, is that now:
 * <kbd>Ctrl-I</kbd><kbd>T</kbd> does **I**dris **t**ype-of;
 * <kbd>Ctrl-I</kbd><kbd>C</kbd> does **I**dris **c**ase-split;
 * <kbd>Ctrl-I</kbd><kbd>W</kbd> does **I**dris make-**w**ith;
 * <kbd>Ctrl-I</kbd><kbd>Enter</kbd> does **I**dris open-repl.

:heavy_check_mark: easy to type
:heavy_check_mark: easy to remember
:heavy_check_mark: no conflicts with system area
:heavy_check_mark: Just Works by default
:heavy_check_mark: #159 fixed

Nothing maps Ctrl-I by default in my Atom installation. Any intra-Atom conflicts, if exist, are unknown to me. Note: this keymap effectively "uses up" *just one* shortcut from the Atom namespace, instead of ~two dozen. This reduces the likelyhood of future conflicts.

The OS X part of keymap, natually, uses <kbd>Cmd-I</kbd> prefix as well, for partly the same reasons as above + keeping consistency.

---
Lastly, I'll tag #26 here for its relevancy.

### CODE REVIEW

**Comment by ulidtko** [comment]
**TL;DR:** just read the diff and hit Merge if LGTY.

**Comment by melted** [comment]
On one hand it's a good idea, On the other hand it will always cause trouble to change such things. 

It will impact @edwinb's book too. Maybe it could be activated by a toggle in the settings page?

**Comment by ulidtko** [comment]
@melted 
> activated by a toggle in the settings page

No, please no. Having an unconflicting, ergonomic and straightforwardly mnenomic keymap "out of the box" is not a preference -- it's an *expectation*.

How otherwise would you suggest, #159 should be fixed?  
The only option besides changing the defaults, is forcing users to patch up defaults' defects  themselves. Which is, kinda, not cool. Not the way it should be done.

----
I do have an idea how to avoid breaking books (as well as blog posts etc): we could bind the Ctrl-Alt-_  shortcuts all to the same action, let's say `language-idris:legacy-keymap-notice`. That would only show an infobox notifying the user about the keymap change, the new shortcuts, and also featuring a button (or a snippet to paste into `keymap.cson`) to the effect of "bring it back how it was before".

Sounds good?

If no objections, I can implement that, no problem. The migration will be extra-smooth then.

@archaeron, can I get your opinion? I see you were working on exactly the same issue in #26. Like my proposal? Would merge / would not merge?

**Comment by ulidtko** [comment]
I've put together a little migration helper UI as above. Looks like this:

screenshot from 2018-05-01 21-59-52

The `migrations.coffee` module is not top-quality, as it's intended to be thrown away sooner or later.

Please review & test.

**Comment by melted** [comment]
Thanks, I’ll look at it after work.

**Comment by justjoheinz** [comment]
I will try out the branch as well, but for me it looks fairly cumbersome. Ctrl-Alt feels like one key for me, but maybe I am just an old dog refusing to learn new tricks.

**Comment by justjoheinz** [comment]
What are the actual keybindings of the emacs idris mode? It would be nice to have something in line with them, in order to help users use both modes.

**Comment by ulidtko** [comment]
@justjoheinz No idea about the emacs-space, unfortunately I can't use emacs (yet).

> Ctrl-Alt feels like one key for me

Well, yeah, maybe. It's still the same three keypresses basically.

The problem with Ctrl-Alt is that Ctrl-Alt-T will pop up a terminal on Ubuntu, and apps like Atom have no chance to bind it -- it's simply not being delivered to apps.

Ctrl-Alt shortcuts are also more likely to clash with other Atom packages, with the likelihood going up as the number of bound Ctrl-Alt-letters grows. Not so with the Ctrl-i prefix.

**Comment by ulidtko** [comment]
Anyone?..

**Comment by melted** [comment]
Sorry for disappearing , I got insanely busy with GDPR. But that will be
over after this week.

Den tir. 22. maj 2018 12.16 skrev Maxim Ivanov <notifications@github.com>:
